cl/corpus_importer/tasks.py

```diff
@@ -1504,15 +1505,27 @@ def download_pacer_pdf_by_rd(
     :param pacer_doc_id: The internal PACER document ID to download
     :param cookies: A requests.cookies.RequestsCookieJar with the cookies of a
     logged-in PACER user.
+    :param magic_number: The magic number to fetch PACER documents for free
+    this is an optional field, only used by RECAP Email documents
     :return: requests.Response object usually containing a PDF, or None if that
     wasn't possible.
     """
+
     rd = RECAPDocument.objects.get(pk=rd_pk)
     pacer_court_id = map_cl_to_pacer_id(rd.docket_entry.docket.court_id)
     s = PacerSession(cookies=cookies)
     report = FreeOpinionReport(pacer_court_id, s)
     try:
-        r = report.download_pdf(pacer_case_id, pacer_doc_id)
+        if magic_number:
+            # If a magic_number is passed as an argument try to download the
+            # document anonymously by magic link
+            de_seq_num = rd.docket_entry.pacer_sequence_number
+            r = report.download_pdf_magic_link(
+                pacer_case_id, pacer_doc_id, de_seq_num, magic_number
+            )
+        else:
+            # If no magic_number passed use normal method to fetch the document
+            r = report.download_pdf(pacer_case_id, pacer_doc_id)
```

Comment:
I'm surprised here that you can't just use something like:

    r = report.download_pdf(pacer_case_id, pacer-doc_id, magic_number)

The one missing piece is the de_seq_num, but is that necessary? If so, could we add it to the `download_pdf` method more easily than adding a whole other method for it in juriscraper?

Reply:
Changed!